### PR TITLE
Use gulp to build dependencies

### PR DESCRIPTION
**Pull request by gagnieray**
All the obsolete dependencies in the sources have been removed.
Some were replaced with ones from the npm registry when necessary.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  './node_modules/leaflet-legend/leaflet-legend.js'
+];
+
+const main_assets = [
+  {
+    'src': './node_modules/leaflet/dist/images/*',
+    'dest': '/images/'
+  },
+  {
+    'src': './node_modules/leaflet.locatecontrol/*.svg',
```
Comment (gagnieray) [inline gulpfile.js:40]: Although svg files are stored in the plugin's images folder and their path later replaced in the css files with the new one, trying to reach them throws an error 😕

```
Type : Error
Code : 0
Message : Call to undefined method DI\Container::halt()
Fichier : [...]/galette/galette/includes/routes/plugins.routes.php
Ligne : 71 
```

Comment (trasher) [reply gulpfile.js:40]: This is an error in Galette core; the `svg` extension is not known. To make this work, you just need to add the extension in the `$auth_ext` array (`galette/includes/routes/plugins.routes.php`).
I'll have to take a look for the `halt` replacement, as far as I remmeber, this method is no longer part of Slim for a while.

Comment (gagnieray) [reply gulpfile.js:40]: Simple as that !
I should have followed the line 😅

File: bin/release [hunk 2]
```diff
@@ -317,17 +317,20 @@ def add_libs(rel_name, galette_archive):
     p1 = subprocess.Popen(npm_cmd, shell=True, cwd=npm_dir)
     p1.wait()
 
-    wp_cmd = 'npm run-script build-dist'
+    wp_cmd = 'gulp'
```
Comment (trasher) [inline bin/release]: When running the release script, I got the following error:
```
/bin/sh: ligne 1: gulp : commande introuvable
```

I do not understand if there is some dependency missing or else; trying to run `npm run-script gulp` (the command used for galette dependencies install fails with the same error message.

Comment (trasher) [reply bin/release]: I think I got it... In package.json file, the script name for the gulp command is `build`. 
```suggestion
    wp_cmd = 'build'
```

Comment (gagnieray) [reply bin/release]: Indeed, this command supposes gulp is installed globally and uses the default function defined in gulpfile.js.

In the release script, the command should actually be set to `npm run build`.

And commands of the scripts in the package.json file should run gulp from `./node_modules/.bin/`.

**Pull request merged**